### ISSUE TITLE
The recent (1.0?) Helm Chart Namespace changes have made the chart unusable

### ISSUE THREAD

**Issue by elimumford** [Opster/opensearch-k8s-operator#132]
The Chart outputs a namespace definition with no name now... and aside from dropping the chart as a raw template and manipulating it, you can no longer install the operator using the Helm Repo at https://opster.github.io/opensearch-k8s-operator-chart/.  These changes all seem to have been merged in the last 24 hours.  PR #130 even references this issue, yet, that PR has not been merged yet.  Seems the changes were made to main and the gh-pages publication process may have 'jumped the gun'.

**Comment by idanl21** [comment]
Hey :)
can you please share the error that you get form helm install ?

**Comment by idanl21** [comment]
hey :)
should be fine now, please try again to install our new v1.0 version and let us know!
thanks for you interest !

**Comment by elimumford** [comment]
Ok, so something has been updating the charts as the namespace resource has disappeared... although I can't see where in any commits to this repo may have caused that (main/1.0 tag both still seem to have it).

The chart will now install (with new output as well...), but new issues related to namespaces again... 
`helm template opensearch-operator opensearch-operator/opensearch-operator --namespace opensearch-operator` results in the Service Account and Operator deployments being launched in the given namespace, but the template outputs ClusterRoleBinding that bind to subjects in the 'default' namespace... 

```yaml
# Source: opensearch-operator/templates/opensearch-operator-manager-rolebinding-crb.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: opensearch-operator-manager-rolebinding
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: opensearch-operator-manager-role
subjects:
- kind: ServiceAccount
  name: opensearch-operator-controller-manager
  namespace: default
---
# Source: opensearch-operator/templates/opensearch-operator-proxy-rolebinding-crb.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: opensearch-operator-proxy-rolebinding
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: opensearch-operator-proxy-role
subjects:
- kind: ServiceAccount
  name: opensearch-operator-controller-manager
  namespace: default
---
```

Which then means the controller can't see anything it needs to... because it's SA doesn't actually have a role bound to it...
` kubectl logs -n opensearch-operator pod/opensearch-operator-controller-manager-67bcfc657b-zfwln -c operator-controller-manager` ==>
```
W0510 17:00:00.205490       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:167: failed to list *v1.Deployment: deployments.apps is forbidden: User "system:serviceaccount:opensearch-operator:opensearch-operator-controller-manager" cannot list resource "deployments" in API group "apps" at the cluster scope
E0510 17:00:00.205517       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:167: Failed to watch *v1.Deployment: failed to list *v1.Deployment: deployments.apps is forbidden: User "system:serviceaccount:opensearch-operator:opensearch-operator-controller-manager" cannot list resource "deployments" in API group "apps" at the cluster scope
W0510 17:00:02.197337       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:167: failed to list *v1.Secret: secrets is forbidden: User "system:serviceaccount:opensearch-operator:opensearch-operator-controller-manager" cannot list resource "secrets" in API group "" at the cluster scope
E0510 17:00:02.197364       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:167: Failed to watch *v1.Secret: failed to list *v1.Secret: secrets is forbidden: User "system:serviceaccount:opensearch-operator:opensearch-operator-controller-manager" cannot list resource "secrets" in API group "" at the cluster scope
W0510 17:00:03.067647       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:167: failed to list *v1.ConfigMap: configmaps is forbidden: User "system:serviceaccount:opensearch-operator:opensearch-operator-controller-manager" cannot list resource "configmaps" in API group "" at the cluster scope
E0510 17:00:03.067674       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:167: Failed to watch *v1.ConfigMap: failed to list *v1.ConfigMap: configmaps is forbidden: User "system:serviceaccount:opensearch-operator:opensearch-operator-controller-manager" cannot list resource "configmaps" in API group "" at the cluster scope
W0510 17:00:03.083552       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:167: failed to list *v1.Pod: pods is forbidden: User "system:serviceaccount:opensearch-operator:opensearch-operator-controller-manager" cannot list resource "pods" in API group "" at the cluster scope
E0510 17:00:03.083575       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:167: Failed to watch *v1.Pod: failed to list *v1.Pod: pods is forbidden: User "system:serviceaccount:opensearch-operator:opensearch-operator-controller-manager" cannot list resource "pods" in API group "" at the cluster scope
W0510 17:00:04.373169       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:167: failed to list *v1.OpenSearchCluster: opensearchclusters.opensearch.opster.io is forbidden: User "system:serviceaccount:opensearch-operator:opensearch-operator-controller-manager" cannot list resource "opensearchclusters" in API group "opensearch.opster.io" at the cluster scope
E0510 17:00:04.373203       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:167: Failed to watch *v1.OpenSearchCluster: failed to list *v1.OpenSearchCluster: opensearchclusters.opensearch.opster.io is forbidden: User "system:serviceaccount:opensearch-operator:opensearch-operator-controller-manager" cannot list resource "opensearchclusters" in API group "opensearch.opster.io" at the cluster scope
```

**Comment by swoehrl-mw** [comment]
Hi @elimumford. I can confirm your observation. Looks like without an explicit namespace defined for the subject of the clusterolebinding kubernetes will use the default namespace instead of the one the helm chart is installed into.

**Comment by swoehrl-mw** [comment]
Should already be fixed in main thanks to @prudhvigodithi and #148
